common/src/main/java/io/crate/interval/PGIntervalParser.java

```diff
@@ -0,0 +1,148 @@
+            final StringTokenizer st = new StringTokenizer(value);
+            for (int i = 1; st.hasMoreTokens(); i++) {
+                String token = st.nextToken();
+
+                if ((i & 1) == 1) {
+                    int endHours = token.indexOf(':');
+                    if (endHours == -1) {
+                        valueToken = token;
+                        continue;
+                    }
+                    // This handles hours, minutes, seconds and microseconds for
+                    // ISO intervals
+                    int offset = (token.charAt(0) == '-') ? 1 : 0;
+
+                    hours = nullSafeIntGet(token.substring(offset + 0, endHours));
+                    minutes = nullSafeIntGet(token.substring(endHours + 1, endHours + 3));
+
+                    int endMinutes = token.indexOf(':', endHours + 1);
+                    seconds = parseInteger(token.substring(endMinutes + 1));
+                    milliSeconds = parseMilliSeconds(token.substring(endMinutes + 1));
+
+                    if (offset == 1) {
+                        hours = -hours;
+                        minutes = -minutes;
+                        seconds = -seconds;
+                        milliSeconds = -milliSeconds;
+                    }
+                    valueToken = null;
+                } else {
+                    // This handles years, months, days for both, ISO and
+                    // Non-ISO intervals. Hours, minutes, seconds and microseconds
+                    // are handled for Non-ISO intervals here.
+                    if (token.startsWith("year")) {
+                        years = nullSafeIntGet(valueToken);
+                    } else if (token.startsWith("mon")) {
+                        months = nullSafeIntGet(valueToken);
+                    } else if (token.startsWith("day")) {
+                        days = nullSafeIntGet(valueToken);
+                    } else if (token.startsWith("week")) {
+                        days = nullSafeIntGet(valueToken) * 7;
+                    } else if (token.startsWith("hour")) {
+                        hours = nullSafeIntGet(valueToken);
+                    } else if (token.startsWith("min")) {
+                        minutes = nullSafeIntGet(valueToken);
+                    } else if (token.startsWith("sec")) {
+                        seconds = parseInteger(valueToken);
+                        milliSeconds = parseMilliSeconds(valueToken);
+                    }
+                }
+            }
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("Invalid interval format " + value);
+        }
+
+        if (years == null && months == null && days == null && hours == null && minutes == null && seconds == null) {
+            throw new IllegalArgumentException("Invalid interval format " + value);
+        }
+
+        Period period = new Period(setToDefault(years),
```

Comment:
Why not initialize the variables to 0 here https://github.com/crate/crate/pull/8927/files#diff-6a7f955a155a6611ff9c41c555c4f9eeR51 ? I think I'm missing something?

Reply:
The values are null when no data was parsed, then it is not a valid interval while the values of an interval can still be 0.

Reply:
It is cleaned up now.